utils/src/main/kotlin/Utils.kt

```diff
@@ -230,16 +229,13 @@ fun normalizeVcsUrl(vcsUrl: String): String {
     //     [scheme:][//authority][path][?query][#fragment]
     // where a server-based "authority" has the syntax
     //     [user-info@]host[:port]
-    val uri = try {
+    val uri = runCatching {
```

Comment:
A significant part of these issues seems to be related to the construction of possibly invalid URIs. Would it make sense to have a special utility function for this purpose, maybe something like this:
`fun <T> withUri(uriStr: String, defaultValue: T?, block: (URI) -> T): T?`
(The signature may need some fine-tuning.)

Reply:
I was briefly thinking about something like this, too, but did not quickly find a proper signature for such a helper function. I'll probably do it as a follow-up.

Reply:
See https://github.com/oss-review-toolkit/ort/pull/3752.